### ISSUE TITLE
Contract bytecode size is extremely large with hashmaps of struct with large array members

### ISSUE THREAD

**Issue by skellet0r** [vyperlang/vyper#2330]
### Version Information

* vyper Version: 0.2.11
* OS: linux
* Python Version : 3.8.6

### What's your issue about?

![Example Bytecode Issue](https://user-images.githubusercontent.com/23556333/111024529-42e7cb00-83ad-11eb-805e-da66663278f2.png)

The issue is contracts that use a struct (with a large size array member) in a hashmap see their bytecode size increase dramatically. Using `brownie compile -s` one can see the bytecode size of compiled contracts.

**Comment by fubuloubu** [comment]
Probably something to do with a missing optimization of return data pointers for statically sized structs

**Comment by charles-cooper** [comment]
The issue is that the two contracts have different semantics. The former contract takes a single argument (an address) generates a return statement which writes the entire struct at the location `data[arg0:address]` to the returndata. The latter takes two arguments (an address and an index into the array) and returns the item at the location `data[arg0:address,arg1:uint256]` to the returndata.

I don't think this is a performance bug per se, but rather the bug is that the semantics are inconsistent or at least unclear. So we need to figure out which one we want.

(for reference, I'm pasting the two contracts here:
```python
# repros/2330.1.vy 
struct Foo:
  value: uint256
  value2: uint256
  value3: uint256
  value4: uint256[256]
data: public(HashMap[address, Foo])
```
```python
# repros/2330.2.vy 
values: public(HashMap[address, uint256])
values2: public(HashMap[address, uint256])
values3: public(HashMap[address, uint256])
values4: public(HashMap[address, uint256[256]])

```

**Comment by charles-cooper** [comment]
the codesize blowup would be solved in https://github.com/vyperlang/vyper/pull/2738

**Comment by charles-cooper** [comment]
re. the differing semantics of the getters, i'm declaring as "wontfix".

the codesize blowup is solved as of 5dc3ac7ec700d85886eda3d53a03abcf5c7efc9c (#3483).
